cylc/flow/workflow_files.py

```diff
@@ -1352,8 +1352,15 @@ def install_workflow(
     # check source link matches the source symlink from workflow dir.
     cylc_install.mkdir(parents=True, exist_ok=True)
     if not source_link.exists():
+        if source_link.is_symlink():
+            raise WorkflowFilesError(
+                'Workflow run directory contains a broken source symlink: '
+                f'{source_link}\n'
+                f'Restore the original source or modify this'
+                ' symlink, then try cylc install again.'
+            )
```

Comment:
I feel like the fact that the symlink is broken shouldn't be the main point, it's that `Workflow source dir is not accessible` (from the other error message). In fact, I think the other error message can be reused for this?

It makes me wonder if we should have a cli option for `cylc install` to "re-source" the workflow from the given directory (which would be a follow-up issue).